### PR TITLE
Pathfinding Enemies

### PR DESCRIPTION
**Pull request by dairykillsme**
Adds enemy pathfinding. I'll  write up more here later because I think this is sufficiently complex to require code review, this is mostly to see how bad the merge is.

### CODE REVIEW

File: Prefabs/Arrow.tscn
Comment (oliveman521) [inline Prefabs/Arrow.tscn:1]: All terrain objects should be layer 1 by default, that way the default collision mask will give predictable results. Player is layer 2. Areas like spikes can still be layer 1, but of course aren't bodies so won't be treated as solid

File: Scripts/Enemy.gd
Comment (oliveman521) [inline Scripts/Enemy.gd:1]: This already has and will continue to desync with the player code. Probably fine, but something to watch out for

File: Levels/World1/temple_1_intro.tscn
Comment (oliveman521) [inline Levels/World1/temple_1_intro.tscn:1]: This probably shouldnt be pushed lol

File: Prefabs/Arrow.tscn
Comment (dairykillsme) [reply Prefabs/Arrow.tscn:1]: Fixed

File: Scripts/GameManager.gd
[the remark this answers is not in the document]
Comment (dairykillsme) [reply Scripts/GameManager.gd:1]: Fixed

File: Levels/World1/temple_1_intro.tscn
Comment (dairykillsme) [reply Levels/World1/temple_1_intro.tscn:1]: Fixed.

**Pull request merged**